homework16/js/main.js

```diff
@@ -0,0 +1,211 @@
+const API_KEY = '91d38c9e';
+const searchInput = document.querySelector('.search-films__input');
+const searchButton = document.querySelector('.search-films__button');
+let pageNum;
+
+function notFoundError(data){
+    const p = document.createElement('div');
+    p.innerText = `${data.Error}`;
+    p.className = 'search-error';
+    document.querySelector('.list-films').append(p);
+}
+
+//Показ фільмів
+function showMovie(film){
+    const divFilmDescription = document.createElement('div');
+    divFilmDescription.className = 'list-films__element';
+
+    //Додаємо зображення
+    const poster = document.createElement('img');
+    if(film.Poster === 'N/A'){
+        poster.src = '../assets/poster.jpg';
+    } else{
+        poster.src = `${film.Poster}`;
+    }
+    poster.alt = `Poster of "${film.Title}"`;
+
+    const divFilmInfo = document.createElement('div');
+    divFilmInfo.className = 'list-films__info';
+
+    const titleOfFilm = document.createElement('p');
+    titleOfFilm.className = 'list-films__description';
+    titleOfFilm.innerText = `${film.Title}, ${film.Year}, ${film.Type}`
+    divFilmInfo.append(titleOfFilm);
+
+    const btnDetails = document.createElement('button');
+    btnDetails.className = `list-films__details`
+    btnDetails.innerText = `Details`;
+    btnDetails.id = film.imdbID;
+
+    //Невеличку інформацію групуємо у окремий блок
+    divFilmDescription.append(poster, divFilmInfo, btnDetails);
+    document.querySelector('.list-films').append(divFilmDescription);
+}
+
+//Запит данних та виведення їх
+function movieList() {
+    document.querySelector('.list-films').innerHTML = '';
+
+    const filmName = searchInput.value;
+    const URL = `http://www.omdbapi.com/?apikey=${API_KEY}&s=${filmName}&page=${pageNum}`;
+    const getFilms = fetch(URL);
+    getFilms
+        .then((result) => {
+            return result.json();
+        })
+        .then((data) => {
+            if(data.Response === 'False'){
+                notFoundError(data);
+            } else {
+                data.Search.forEach((film) => {
+                    showMovie(film);
+                });
+
+                //Вивід кнопок пагінації
+                getPaginationNumbers(getPageCount(data.totalResults));
+
+                //Функціонал кнопки Details
+                document.querySelector('.list-films').addEventListener('click', function (event) {
+                    if (event.target.classList.contains('list-films__details')) {
```

Comment:
It isn't a good idea, because `else` will by any click


https://user-images.githubusercontent.com/84010501/204128453-2906216f-c964-48a2-b481-3b07d20122e6.mov